src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -508,7 +511,7 @@ void ConvolutionType<
   arma::Cube<typename MatType::elem_type> tempCube;
   MakeAlias(tempCube, temp.memptr(), apparentWidth, apparentHeight,
       inMaps * higherInDimensions * batchSize);
-  paddingBackward.Backward(input, usingPadding ? inputPadded : input, temp);
+  paddingBackward.Backward(input, {} /* unused */, usingPadding ? inputPadded : input, temp);
```

Comment:
I did have that thought, given that Backward sometimes has that need.  I don't know if it's strictly necessary.  Maybe later?

Reply:
Yeah, totally fine with saving it for the day when it actually turns out to be useful or necessary. :+1: Certainly no need to do it as a part of this PR.

Reply:
Up to you.  I don't mind refactoring `Gradient()` to take the output too (we could open an issue so we don't forget about it).  Does the averaging layer you were working on have application for any widely-used architectures that are already out there?  If it's a one-off, maybe it doesn't make sense to contribute it back, but if it's a building block that will get used, it could be nice to include. :+1: